lib/Doctrine/ORM/Tools/EntityGenerator.php

```diff
@@ -1145,9 +1145,10 @@ protected function generateDiscriminatorColumnAnnotation(ClassMetadataInfo $meta
             return '';
         }
 
-        $columnDefinition = 'name="' . $discrColumn['name']
-            . '", type="' . $discrColumn['type']
-            . '", length=' . $discrColumn['length'];
+        $columnDefinition = sprintf('name="%s", type="%s"', $discrColumn['name'], $discrColumn['type']);
+        if (isset($discrColumn['length'])) {
```

Comment:
length might not be defined.
I assume this was always used with string so no error was reporter.